### PR TITLE
site24x7 apminsight npm package type declaration changes

### PR DESCRIPTION
**Pull request by pawan-yadav-2008**
Please fill in this template.

- [ ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/apminsight/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
+// Type definitions for apminsight 3.0
+// Project: https://www.npmjs.com/package/apminsight
+// Definitions by: Site24x7 <https://github.com/site24x7>
```
Comment (peterblazejewicz) [inline types/apminsight/index.d.ts:3]: this cannot be an organization, the owner is a (technically) Github handle, with a person behind this, sorry!

**Review by jakebailey — APPROVED** [review]

**Comment by peterblazejewicz** [comment]
IMO this needs @selvishree0630 ACK, being added in the header (not the PR owner)

**Review by jakebailey — CHANGES_REQUESTED** [review]
Oops, didn't notice.

**Comment by selvishree0630** [comment]
> IMO this needs @selvishree0630 ACK, being added in the header (not the PR owner)

Ready to merge

**Review by sandersn — APPROVED** [review]

**Review by sandersn — CHANGES_REQUESTED** [review]
Looking at the apminsight package, it looks like you publish that package as well. The better way to publish types is to include them in apminsight so that people only have to install 1 package. Otherwise, they'll have to install `apminsight` and `@types/apminsight`.

In other words, DT is intended for third parties to publish types for packages they don't control. If you publish a package, you should include the types in the package.

**Comment by selvishree0630** [comment]
> Looking at the apminsight package, it looks like you publish that package as well. The better way to publish types is to include them in apminsight so that people only have to install 1 package. Otherwise, they'll have to install `apminsight` and `@types/apminsight`.
> 
> In other words, DT is intended for third parties to publish types for packages they don't control. If you publish a package, you should include the types in the package.

Yes I agree. We have also added declaration file in our package and it will be released soon. But, if our users have lower version of node or npm, it may fail. So as an alternate option, we want to publish our package here also.

**Comment by sandersn** [comment]
Typescript is the tool that looks up types, not node or npm. But Typescript resolves types in `@types` and `package.json` the same way. That feature has been available since 2.0 released in 2016, 7 years ago.

Are you talking about old versions of `apminsight`? It could make sense to publish types for versions 1 and 2 of apminsight on DT. But the PR currently says these are 3's types.

**Comment by selvishree0630** [comment]
> Typescript is the tool that looks up types, not node or npm. But Typescript resolves types in `@types` and `package.json` the same way. That feature has been available since 2.0 released in 2016, 7 years ago.
> 
> Are you talking about old versions of `apminsight`? It could make sense to publish types for versions 1 and 2 of apminsight on DT. But the PR currently says these are 3's types.

We will be releasing the apminsight package with types as v4.0. We need to publish our package v3.0 in `@types` since users using v3.x may need it. And types for versions 1 and 2 will not be needed. We will ask users to use either version 3 or 4.

**Comment by sandersn** [comment]
Is it possible to publish a new version of 3.* that includes the types you have in this PR? It seems like the 3.* branch is still active so I guess you might be able to publish new versions of the package.

If it's not feasible then we can publish the types here but it's not ideal because the usual thing to do with packages that publish newer versions with types is to **delete** the old versions from DT. So the path would be (1) put the 3.* types on DT (2) publish 4.0 shortly afterward (3) delete the 3.* types from DT, leaving `@types/apminsight@3` on npm.

**Comment by selvishree0630** [comment]
> Is it possible to publish a new version of 3.* that includes the types you have in this PR? It seems like the 3.* branch is still active so I guess you might be able to publish new versions of the package.
> 
> If it's not feasible then we can publish the types here but it's not ideal because the usual thing to do with packages that publish newer versions with types is to **delete** the old versions from DT. So the path would be (1) put the 3.* types on DT (2) publish 4.0 shortly afterward (3) delete the 3.* types from DT, leaving `@types/apminsight@3` on npm.

Okay then. We will publish the types with the apminsight package itself since releasing it here will not be needed. We will close the pull request. Thank you for the clarifications.

**Pull request closed without merging**